### PR TITLE
17573: DebtRes FSR add wizard page

### PR DESCRIPTION
**Pull request by craigwheeler**
## Description
adds wizard page to FSR refer to the mock: https://preview.uxpin.com/803dc24213cf723bfce6c5e3eaef5a2c3f3e344b#/pages/133701647/simulate/sitemap?mode=i

## Screenshots
Financial Status Report  Veterans Affairs 2021-03-03 11-42-27

## Definition of done
- [x] Changes have been tested in vets-website
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/financial-status-report/containers/App.jsx [hunk 1]
```diff
@@ -23,24 +37,24 @@ const App = ({
     [getFormStatus],
   );
 
+  if (wizardState !== WIZARD_STATUS_COMPLETE) {
+    return <WizardContainer setWizardStatus={setWizardStatus} />;
+  }
+
```
Comment (LinnJS) [inline src/applications/financial-status-report/containers/App.jsx:43]: Love the if statements vs all the conditionals way more readable!

File: src/applications/financial-status-report/wizard/pages/Decision.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  {
+    value: 'wrong',
+    label: 'I think the amout of this debt is wrong.',
+  },
+];
+
+const Decision = ({ setPageState, state = {} }) => {
+  const setState = ({ value }) => {
+    switch (value) {
```
Comment (LinnJS) [inline src/applications/financial-status-report/wizard/pages/Decision.jsx:23]: Great use of a switch statement have been thinking a lot about switch vs ifs lately perfect use case

File: src/applications/financial-status-report/wizard/pages/Error.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          <a href="https://iris.custhelp.va.gov/app/ask">
+            Go to our online question form (called IRIS)
+          </a>
+          . On the IRIS page, select <strong>Debt Management Center</strong>,
+          your debt type, and <strong>Dispute</strong> within the Topic
+          dropdown. For Inquiry Type, select <strong>Question</strong>.
+        </li>
+        <li>
+          <strong>Mail: </strong>
+          <div>Debt Management Center</div>
```
Comment (LinnJS) [inline src/applications/financial-status-report/wizard/pages/Error.jsx:30]: super nit pick but I would probably use `<span>` tags here

File: src/applications/financial-status-report/wizard/pages/LessThanFive.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        Call us at <Telephone contact={'800-827-0648'} /> (or{' '}
+        <Telephone contact={'1-612-713-6415'} /> from overseas). We’re here
+        Monday through Friday, 7:30 a.m. to 7:00 p.m. ET. If you have hearing
+        loss, call TTY:{' '}
+        <Telephone contact={CONTACTS[711]} pattern={PATTERNS['3_DIGIT']} />.
+      </li>
+      <li>
+        <strong>Mail: </strong>
+        <div>Debt Management Center</div>
+        <div>P.O. Box 11930</div>
```
Comment (LinnJS) [inline src/applications/financial-status-report/wizard/pages/LessThanFive.jsx:46]: same here but totally optional maybe there is a reason

**Review by LinnJS — APPROVED** [review]
Looks like mostly content and adding some functionality where business logic already made in redux actions. Some minor optional code style comments. Code look super readable and clean!

File: src/applications/financial-status-report/wizard/pages/Dependents.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        <Telephone contact={'1-612-970-5688'} />
+      </li>
+    </ul>
+    <p>
+      <strong>If you submitted VA Form 5655 in the past 6 months</strong>
+    </p>
+    <p>
+      You don’t need to submit a new request unless you have changes to report.
+      Call us at <Telephone contact={'800-827-0648'} /> (or{' '}
+      <Telephone contact={'1-612-713-6415'} /> from overseas). We’re here Monday
```
Comment (Mottie) [inline src/applications/financial-status-report/wizard/pages/Dependents.jsx:49]: There is an `OUTSIDE_US` pattern that should be added here - https://github.com/department-of-veterans-affairs/component-library/blob/master/src/components/Telephone/Telephone.jsx#L12

File: src/applications/financial-status-report/wizard/pages/Disagree.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        you can submit a supplemental claim or request a higher level review or
+        board appeal.
+        <a className="wizard-links" href="/decision-reviews/">
+          Learn more about the VA appeals process
+        </a>
+      </p>
+      <p>
+        <strong>If you need more help, </strong>
+        call your VA benefit office.
+        <a className="wizard-links" href="/resources/helpful-va-phone-numbers/">
```
Comment (Mottie) [inline src/applications/financial-status-report/wizard/pages/Disagree.jsx:23]: Is this `wizard-links` class for all links? I didn't see it in any of the other links on other wizard pages.

File: src/applications/financial-status-report/wizard/pages/LessThanFive.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        </a>
+        . On the IRIS page, select <strong>Debt Management Center</strong>, your
+        debt type, and <strong>Payment Plan</strong> within the Topic dropdown.
+        For Inquiry Type, select <strong>Question</strong>. Write your request
+        within the <strong>Question</strong> section.
+      </li>
+      <li>
+        <strong>Phone: </strong>
+        Call us at <Telephone contact={'800-827-0648'} /> (or{' '}
+        <Telephone contact={'1-612-713-6415'} /> from overseas). We’re here
```
Comment (Mottie) [inline src/applications/financial-status-report/wizard/pages/LessThanFive.jsx:38]: `OUTSIDE_US` pattern here too

File: src/applications/financial-status-report/wizard/pages/Submit.jsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        label={label}
+        ariaId={'other_ways_to_file_526'}
+      />
+      <p>
+        <strong>If you submitted VA Form 5655 in the past 6 months</strong>
+      </p>
+      <p>
+        You don’t need to submit a new request unless you have changes to
+        report. Call us at <Telephone contact={'800-827-0648'} /> (or{' '}
+        <Telephone contact={'1-612-713-6415'} /> from overseas). We’re here
```
Comment (Mottie) [inline src/applications/financial-status-report/wizard/pages/Submit.jsx:29]: `OUTSIDE_US` again 😺

**Review by Mottie — CHANGES_REQUESTED** [review]
Overall it looks good. Main thing that needs addressing is the a11y color contrast issue.

File: src/applications/financial-status-report/containers/App.jsx [hunk 9]
```diff
@@ -23,24 +37,24 @@ const App = ({
...
+  if (pending) {
+    return <LoadingIndicator setFocus message="Loading your information..." />;
+  }
+
+  if (isLoggedIn && !pending && isError) {
```
[the remark this answers is not in the document]
Comment (craigwheeler) [reply src/applications/financial-status-report/containers/App.jsx:48]: oh ya your right i can clean up this logic a little better thanks

File: src/applications/financial-status-report/components/ErrorMessage.jsx [hunk 10]
```diff
@@ -3,25 +3,23 @@ import { connect } from 'react-redux';
...
+  return (
+    errorCode === 'FSR_SERVER_ERROR' && (
+      <div className="row vads-u-margin-bottom--3">
+        <AlertBox
+          status="error"
+          headline="We're sorry. Something went wrong on our end."
+        >
+          <div>
+            <p>
+              You can't submit a Financial Status Report (5655) because
```
[the remark this answers is not in the document]
Comment (craigwheeler) [reply src/applications/financial-status-report/components/ErrorMessage.jsx:15]: Theres been a lot of discussion with content around FSR and this is how it is currently on the mocks but im sure will change eventually

File: src/applications/financial-status-report/sass/financial-status-report.scss [hunk 11]
```diff
@@ -167,6 +167,23 @@
...
 
+// wizard
+.fsr-wizard {
+  .wizard-content {
+    margin: 30px 0;
+    .fieldset-input:first-child {
+      margin-top: 0;
+    }
+    .separation-pay {
+      color: $color-green;
```
[the remark this answers is not in the document]
Comment (craigwheeler) [reply src/applications/financial-status-report/sass/financial-status-report.scss:178]: ya this did seem a off to me as well but using what was on the mocks but ill confirm this with UI thanks

File: src/applications/financial-status-report/wizard/pages/Dependents.jsx (hunk 5 above)
Comment (craigwheeler) [reply src/applications/financial-status-report/wizard/pages/Dependents.jsx:49]: oh ya meant to go back on this thanks ill update it

File: src/applications/financial-status-report/wizard/pages/Disagree.jsx (hunk 6 above)
Comment (craigwheeler) [reply src/applications/financial-status-report/wizard/pages/Disagree.jsx:23]: it was meant to be a class to style some links throughout the wizard but may possibly not be needed ill look into it tks

File: src/applications/financial-status-report/sass/financial-status-report.scss (hunk 11 above)
[the remark this answers is not in the document]
Comment (craigwheeler) [reply src/applications/financial-status-report/sass/financial-status-report.scss:178]: this was actually a mistake on the mock and should have been bold without the use of color.. good catch tks!

[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/financial-status-report/sass/financial-status-report.scss:183]: You can remove this definition if you aren't using it 😸

**Review by Mottie — APPROVED** [review]
ghe-emoji

**Pull request merged**